egcg_core/rest_communication.py

```diff
@@ -139,11 +139,16 @@ def _req(self, method, url, quiet=False, **kwargs):
                 raise RestCommunicationError('Cannot upload files and nested json in one query')
             kwargs['data'] = kwargs.pop('json')
 
-        self.lock.acquire()
         try:
-            r = self.session.request(method, url, **kwargs)
-        finally:
-            self.lock.release()
+            with self.lock:
+                r = self.session.request(method, url, **kwargs)
+        except Exception as e:
```

Comment:
Is this too broad? The urllib3 Retry catches TimeoutError, HTTPException, SocketError, ProtocolError, BaseSSLError, SSLError and CertificateError.

Reply:
Because we would eventually raise it, I think it would be fine